src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -806,26 +804,53 @@ private Task AssignVmAsContributorToStorageAccountAsync(IIdentity managedIdentit
                 async () =>
                 {
-                    var storageAccount = await azureClient.StorageAccounts
-                        .Define(configuration.StorageAccountName)
-                        .WithRegion(configuration.RegionName)
-                        .WithExistingResourceGroup(configuration.ResourceGroupName)
-                        .CreateAsync(cts.Token);
+                    var storageAccount = await TryGetExistingStorageAccountAsync() ?? await CreateNewStorageAccountAsync();
 
                     cts.Token.ThrowIfCancellationRequested();
-
                     var blobClient = await GetBlobClientAsync(storageAccount);
-                    var defaultContainers = new List<string> { WorkflowsContainerName, InputsContainerName, "cromwell-executions", "cromwell-workflow-logs", "outputs", ConfigurationContainerName };
 
+                    var defaultContainers = new List<string> { WorkflowsContainerName, InputsContainerName, "cromwell-executions", "cromwell-workflow-logs", "outputs", ConfigurationContainerName };
                     await Task.WhenAll(defaultContainers.Select(c => blobClient.GetContainerReference(c).CreateIfNotExistsAsync(cts.Token)));
 
                     return storageAccount;
                 });
         }
 
+        private async Task<IStorageAccount> CreateNewStorageAccountAsync()
+        {
+            return await azureClient.StorageAccounts
+                                        .Define(configuration.StorageAccountName)
+                                        .WithRegion(configuration.RegionName)
+                                        .WithExistingResourceGroup(configuration.ResourceGroupName)
+                                        .WithOnlyHttpsTraffic()
+                                        .CreateAsync(cts.Token);
+        }
+
+        private async Task<IStorageAccount> TryGetExistingStorageAccountAsync()
+        {
+            return await TryGetExistingStorageAccountAsync(configuration.StorageAccountName);
+        }
+
+        private async Task<IStorageAccount> TryGetExistingStorageAccountAsync(string storageAccountName)
+        {
+            return (await azureClient.StorageAccounts.ListAsync())
+                .SingleOrDefault(a => a.Name.Equals(storageAccountName, StringComparison.OrdinalIgnoreCase) && a.RegionName.Equals(configuration.RegionName, StringComparison.OrdinalIgnoreCase));
+        }
+
+        private async Task<BatchAccount> TryGetExistingBatchAccountAsync()
+        {
+            return await TryGetExistingBatchAccountAsync(configuration.BatchAccountName);
+        }
+
+        private async Task<BatchAccount> TryGetExistingBatchAccountAsync(string batchAccountName)
+        {
+            return (await new BatchManagementClient(tokenCredentials) { SubscriptionId = configuration.SubscriptionId }.BatchAccount.ListAsync())
+                .SingleOrDefault(a => a.Name.Equals(batchAccountName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Add region (location) condition here. Reason is that name and region uniquely identify the batch account, while storage account name by itself is unique. This way both of these methods will return single account. For storage account, validation checks the existence + region of the returned account. For batch we'd have to return a list of accounts to do the region check in validation, not worth it.

Reply:
I have a check for the region now.